src/classes/OPP_CascadeDeleteLookups_TDTM.cls

```diff
@@ -62,11 +62,12 @@ public with sharing class OPP_CascadeDeleteLookups_TDTM extends CDL_CascadeDelet
         * @return List<Allocation__c> Deleted Opportunities' Allocations.
         ********************************************************************************************************/
         public List<Allocation__c> load(Set<Id> opportunityIds) {
-            return [
-                SELECT Opportunity__c
-                FROM Allocation__c
-                WHERE Opportunity__c IN :opportunityIds OR Payment__r.npe01__Opportunity__c IN :opportunityIds
-            ];
+
```

Comment:
@npsp-reedestockton I talked to @jbrandolinisf about this. He had a great suggestion to ensure that the query/logic is both scalable and handles a possible scenario where PaymentAllocations is enabled in an org and then disabled at a later time.

The change is effectively to:
- Query all Payment records for the set of OpporutnityIds to build a set of PaymentIds
- Modify the Allocation to use:
   - `WHERE Opportunity__c  IN :opportunityIds OR Payment__c IN :paymentIds`

The downside of this is that adds an additional query to the logic. The upside is that it is guaranteed to get any PaymentAllocation records in the event this functionality was enabled and then disabled. Plus, Jon pointed out that the query against `Payment__r.npe01__Opportunity__c` could actually be the root cause of the non-selective query error because it might be causing a table scan. I do feel this is mostly a guessing game for us though. In the mean time I like the idea of getting the paymentIds first because it addresses the possible scenario of PaymentAllocations being disabled in an org after it was enabled, and it it _may_ be more selective. Of course, you may be able to find some additional data in your query tests in the LDV-XXL org once it's online.

Reply:
I am MUCH happier with the two query approach.  We are unlikely to be pushing the limits in this context, so while it will be a little less efficient, it feels a lot less likely to cause issues.

Reply:
Restored to the version @lparrott originally wrote.  I'm happy with this outcome.  Given the location of the non-selective query error, I think there is a fair chance this was the cause of the problem.